lib/fuse_i.h

```diff
@@ -87,6 +87,41 @@ struct fuse_module {
 	int ctr;
 };
 
+/**
+ * Configuration parameters passed to fuse_session_loop_mt() and
+ * fuse_loop_mt().
+ *
+ * Internal API to avoid exposing the plain data structure and
+ * causing compat issues after adding or removing struct members.
+ *
+ */
+struct fuse_loop_config2 {
```

Comment:
Let's keep the name as is (`fuse_loop_config`), and instead introduce `fuse_loop_config_pre311` for the old struct.

Reply:
Yes, I had considered this, but I have a concern here. Basically this will break API compatibility. So applications that are using this struct and that were compiling fine against libfuse <= 3.10 would now fail to compile against libfuse >= 3.11. Take for example Ubuntu/Debian, sometimes all packages get recompiled because package format changes or new package options are set - once libfuse is updated to 3.11 quite some old packages would fail cause mass bug reports. Same for arbitrary non-dev users that just want to compile a random fuse file system that didn't get a recent update. So from my point of view and even though ABI compatibility is kept, keeping "fuse_loop_config" for the new struct would be an libfuse-4 change. 
Hmm, I have an idea how to solve this with ifdef on the version. I need to think about it a bit.

Reply:
Why would they fail to compile if the struct gains an additional member?

Reply:
Well the problem I have is that I really wouldn't like to expose the struct details - I had made struct fuse_loop_config2 private for the reason that we need to ensure that members are correctly initialized. As soon as you just add members there is no enforcement anymore that people use fuse_loop_config_create(), even though the current version 311 is defined - all bad and expected things might come up.